src/components/tl/cuda/alltoallv/alltoallv_ce.c

```diff
@@ -0,0 +1,316 @@
+            return task->super.super.status;
+        }
+        status = ucc_tl_cuda_alltoallv_setup_start(task);
+        if (status != UCC_OK) {
+            task->super.super.status = status;
+            return task->super.super.status;
+        }
+        task->alltoall_ce.stage = ALLTOALL_CE_STAGE_SETUP;
+    case ALLTOALL_CE_STAGE_SETUP:
+        status = ucc_tl_cuda_alltoallv_setup_test(task);
+        if (status != UCC_OK) {
+            task->super.super.status = status;
+            return task->super.super.status;
+        }
+        status = ucc_tl_cuda_alltoallv_ce_post_copies(task);
+        if (ucc_unlikely(status != UCC_OK)) {
+            task->super.super.status = status;
+            return task->super.super.status;
+        }
+        task->alltoall_ce.stage = ALLTOALL_CE_STAGE_COPY;
+    case ALLTOALL_CE_STAGE_COPY:
+        status =
+            ucc_ec_event_test(task->alltoall_ce.copy_done, UCC_EE_CUDA_STREAM);
+        if (status != UCC_OK) {
+            task->super.super.status = status;
+            return task->super.super.status;
+        }
+        status =
+            ucc_tl_cuda_shm_barrier_start(UCC_TL_TEAM_RANK(team), task->bar);
+        if (ucc_unlikely(status != UCC_OK)) {
+            task->super.super.status = status;
+            return task->super.super.status;
+        }
+        task->alltoall_ce.stage = ALLTOALL_CE_STAGE_BAR;
+    default:
+        ucc_assert(task->alltoall_ce.stage == ALLTOALL_CE_STAGE_BAR);
+        break;
+    }
+    task->super.super.status =
+        ucc_tl_cuda_shm_barrier_test(UCC_TL_TEAM_RANK(team), task->bar);
+    if (task->super.super.status == UCC_OK) {
+        ucc_tl_cuda_put_sync(task);
+    }
+    return task->super.super.status;
+}
+
+ucc_status_t ucc_tl_cuda_alltoallv_ce_start(ucc_coll_task_t *coll_task)
+{
+    ucc_tl_cuda_task_t *task = ucc_derived_of(coll_task, ucc_tl_cuda_task_t);
+    ucc_tl_cuda_team_t *team = TASK_TEAM(task);
+
+    task->alltoall_ce.stage = ALLTOALL_CE_STAGE_SYNC;
+    ucc_tl_cuda_alltoallv_ce_progress(coll_task);
+    if (task->super.super.status == UCC_INPROGRESS) {
+        ucc_progress_enqueue(UCC_TL_CORE_CTX(team)->pq, &task->super);
+        return UCC_OK;
+    }
+    return ucc_task_complete(coll_task);
+}
+
+ucc_status_t ucc_tl_cuda_alltoallv_ce_init(ucc_tl_cuda_task_t *task)
+{
+    ucc_tl_cuda_team_t *team = TASK_TEAM(task);
+    ucc_coll_args_t    *args = &TASK_ARGS(task);
+    ucc_status_t        status;
+    size_t              data_len;
+
+    status =
+        ucc_ec_create_event(&task->alltoall_ce.copy_done, UCC_EE_CUDA_STREAM);
+    if (ucc_unlikely(status != UCC_OK)) {
+        return status;
+    }
+
+    data_len = ucc_dt_size(args->src.info_v.datatype) *
+               ucc_coll_args_get_total_count(args, args->src.info_v.counts,
+                                             UCC_TL_TEAM_SIZE(team));
+    status = ucc_tl_cuda_mem_info_get(args->src.info_v.buffer, data_len, team,
```

Comment:
src and dst buffers are not always contig in alltoallv, so mem info get might fail here.
As a quick solution pls add check for UCC_COLL_ARGS_FLAG_CONTIG_SRC_BUFFER and UCC_COLL_ARGS_FLAG_CONTIG_DST_BUFFER and similarly to inplace flag return UCC_ERR_NOT_SUPPORTED

Reply:
Added a check in the beginning of `ucc_tl_cuda_alltoallv_ce_init`.